keyserver/listener.go

```diff
@@ -0,0 +1,127 @@
+package keyserver
+
+import (
+	"encoding/json"
+	"io"
+	"log"
+	"net"
+	"time"
+
+	. "github.com/coniks-sys/coniks-go/protocol"
+)
+
+func (server *ConiksServer) listenForRequests(ln net.Listener, handler func(msg []byte) ([]byte, error)) {
+	defer ln.Close()
+	go func() {
+		<-server.stop
+		ln.Close()
+	}()
+
+	for {
+		conn, err := ln.Accept()
+		if err != nil {
+			select {
+			case <-server.stop:
+				return
+			default:
+				log.Printf("accept client: %s", err)
+				continue
+			}
+		}
+		server.waitStop.Add(1)
```

Comment:
Is this wait necessary, since we're already waiting on `ln.Close()`?  In other words, will any response get through after a call to `ln.Close()`?